### PR TITLE
DSBD-34: read demo cypress e2e test scenarios with a few minor changes.

### PR DESCRIPTION
**Pull request by n3op2**
### What

E2E Integration tests for the **read demo**. This suite covers full user's flow (demo 1a and demo 1b) and responses. It also does minimum check to confirm that right template has been loaded and some errors e.g. password too long.  There are areas to expand on which I left in the comments.

#### Coverage
```
====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        10.4.0                                                                         │
  │ Browser:        Electron 102 (headless)                                                        │
  │ Node Version:   v16.14.2 (/Users/PMichelevicius/.nvm/versions/node/v16.14.2/bin/node)          │
  │                 m                                                                              │
  │ Specs:          1 found (read-demo.spec.js)                                                    │
  │ Searched:       cypress/integration/read-demo.spec.js                                          │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
    Running:  read-demo.spec.js                                                               (3 of 5)


  Read Demo
    Header
      ✓ renders read demo header (458ms)
      ✓ clicking on close button takes to the main menu (482ms)
    Happy path
      ✓ confirms styling and DOM elements Demo 1A (231ms)
      Enters too long password
        ✓ cuts off long passwords (731ms)
      Submits password
        ✓ renders hacker app (608ms)
        ✓ removes password form (610ms)
      Opens hacker app
        ✓ renders modal with yes/no option (753ms)
        Selects [no] option
          ✓ updates state so moddal is no longer rendered (809ms)
      Initiates hacking
        ✓ renders progress animation (2235ms)
        ✓ displays user`s password (10406ms)
        ✓ renders a side button for demo 1b (7690ms)
      Demo 1B
        ✓ confirms styling and DOM elements of Demo 1B (8327ms)
        Submits password
          ✓ renders morello hacker app (8575ms)
          ✓ removes password input (9214ms)
        Opens hacker app
          ✓ renders modal with yes/no option (9660ms)
          Selects [no] option
            ✓ updates state so moddal is no longer rendered (8882ms)
        Initiates hacking
          ✓ renders progress animation (10253ms)
          ✓ renders HACK FAILED (23097ms)
          - details button renders a concole with all output
          ✓ renders Learn More side button (15459ms)

```

### CODE REVIEW

File: cypress/integration/read-demo.spec.js [hunk 1]
```diff
@@ -1,51 +1,268 @@
...
+            .should('exist')
+            .should(
+              'include.text',
+              'Would you like to break into the system and reveal the password?'
+            )
+          cy.get('[data-cy=modal-btn-yes-cheri]')
+            .should('exist')
+            .and('include.text', 'YES')
+            .and('have.css', 'font-family')
+            .and('contain', 'Arial')
```
Comment (jonmattgray) [inline cypress/integration/read-demo.spec.js:217]: are the yes/no buttons in the modal supposed to have `Arial` font? I assumed that was a mistake and fixed it in https://github.com/digicatapult/morello-ui/pull/41 but might be wrong

I think this PR should merge before that font change PR so I can change it to match whichever font is correct

Comment (n3op2) [reply cypress/integration/read-demo.spec.js:217]: To be fair I'm not sure

Comment (n3op2) [reply cypress/integration/read-demo.spec.js:217]: Yea, it already creates conflicts, and I got another 2 ready which will have conflicts as this one did not got merged earlier, well.

**Review by mattdean-digicatapult — APPROVED** [review]
I think this looks all good

**Pull request merged**